### PR TITLE
Add Per-Guild Avatars and ModifyCurrentMember

### PR DESCRIPTION
**Pull request by carlst99**
Started implementing the changes made in [commit d49d34b](https://github.com/discord/discord-api-docs/commit/d49d34bd5ffa3df659eb92bfa743af79c607d5cd) to the DAPI docs.

- [x] Add `CDN.GetGuildMemberAvatarUrl`.
- [x] Add the `Avatar` field to `IGuildMember` and related gateway events.
- [x] Add `IDiscordRestGuildAPI.ModifyCurrentMember`.

Note that there is a caveat here. The DAPI docs note that the new `avatar` field is required, however my current implementation  marks it as optional, as this is the behaviour observed from the API. I have created an [issue on the docs regarding this](https://github.com/discord/discord-api-docs/issues/3883).

I decided to do this in order to have correct functionality of the feature at this time, and because the same workaround has been utilised for role icons.

### CODE REVIEW

File: Backend/Remora.Discord.API.Abstractions/API/Rest/IDiscordRestGuildAPI.cs [hunk 1]
```diff
@@ -335,6 +335,20 @@ Task<Result> ModifyGuildMemberAsync
...
         );
 
+        /// <summary>
+        /// Modifies the current member.
+        /// </summary>
+        /// <param name="guildID">The ID of the guild.</param>
+        /// <param name="nickname">The new nickname.</param>
+        /// <param name="ct">The cancellation token for this operation.</param>
+        /// <returns>A modification result which may or may not have succeeded, containing the updated member.</returns>
+        Task<Result<IGuildMember>> ModifyCurrentMemberAsync
```
Comment (carlst99) [inline Backend/Remora.Discord.API.Abstractions/API/Rest/IDiscordRestGuildAPI.cs:345]: It seems very unusual that this endpoint would not support the `X-Audit-Log-Reason` header. I wonder if this is a mistake on behalf of the docs.

Comment (carlst99) [reply Backend/Remora.Discord.API.Abstractions/API/Rest/IDiscordRestGuildAPI.cs:345]: Testing shows that the endpoint does indeed support the header, and the docs are incorrect.

Comment (n-Ultima) [reply Backend/Remora.Discord.API.Abstractions/API/Rest/IDiscordRestGuildAPI.cs:346]: Probably should open an issue on the docs repo,or I can when I get home from school later.

Comment (carlst99) [reply Backend/Remora.Discord.API.Abstractions/API/Rest/IDiscordRestGuildAPI.cs:346]: I was planning on opening a PR to fix it but haven't gotten around to it yet. If you'd like to do that it'd be great!

File: Backend/Remora.Discord.Rest/API/Guilds/DiscordRestGuildAPI.cs [hunk 2]
```diff
@@ -533,6 +533,26 @@ public virtual Task<Result> ModifyGuildMemberAsync
         }
 
         /// <inheritdoc />
+        public virtual Task<Result<IGuildMember>> ModifyCurrentMemberAsync
```
Comment (Nihlus) [inline Backend/Remora.Discord.Rest/API/Guilds/DiscordRestGuildAPI.cs:536]: Remember to add an override in the caching library as well :)

**Comment by Nihlus** [comment]
Nicely done, thanks!

File: Backend/Remora.Discord.Rest/API/Guilds/DiscordRestGuildAPI.cs (hunk 2 above)
Comment (carlst99) [reply Backend/Remora.Discord.Rest/API/Guilds/DiscordRestGuildAPI.cs:536]: Ah, sure thing!

File: Backend/Remora.Discord.API.Abstractions/API/Rest/IDiscordRestGuildAPI.cs (hunk 1 above)
Comment (carlst99) [reply Backend/Remora.Discord.API.Abstractions/API/Rest/IDiscordRestGuildAPI.cs:346]: @n-Ultima I've gone through and made the PR 👍

Comment (n-Ultima) [reply Backend/Remora.Discord.API.Abstractions/API/Rest/IDiscordRestGuildAPI.cs:346]: I saw when I was at school earlier, let's hope it get's merged easily :)

**Pull request merged**